node/src/components/consensus/protocols/highway/synchronizer.rs

```diff
@@ -395,6 +411,27 @@ impl<I: NodeIdT, C: Context + 'static> Synchronizer<I, C> {
         self.vertices_to_be_added.retain_evidence_only();
     }
 
+    /// Adds `sender` as alternative source for downloading vertex.
+    /// Returns protocol outcomes scheduled as a result of this action.
+    pub(crate) fn alternative_source(
+        &mut self,
+        pvv: PreValidatedVertex<C>,
+        sender: I,
+        now: Timestamp,
+    ) -> ProtocolOutcomes<I, C> {
+        match self
+            .vertices_to_be_added
```

Comment:
Aren't there several other collections, like `vertex_deps`? We can't expect the vertex to be in `vertices_to_be_added`, in general.

Reply:
Reverted in [`81c8391` (#1175)](https://github.com/CasperLabs/casper-node/pull/1175/commits/81c8391fc53aaf2512a9398393fefaba1277d608).